framework/doc/content/documentation/application_development/performance_benchmarking.md

```diff
@@ -1,39 +1,93 @@
 ```
 
-in your boot loader (e.g. grub).  The benchmarking tools should automatically
-detect CPU isolation on Linux and schedule benchmark jobs to those CPUs.You
-should also disable any turbo functionality.  For example on ``intel_pstate``
-driver cpus:
+in your boot loader (e.g. grub).  The benchmarking tools/scripts in MOOSE should automatically
+detect CPU isolation on Linux and schedule benchmark jobs to those CPUs. You should also disable
+any turbo functionality.  For example on `intel_pstate` driver cpus:
 
 ```
 $ echo "1" > /sys/devices/system/cpu/intel_pstate/no_turbo
 ```
 
-This will need to be done on every boot.  You can use the sysfsutils package
-and its ``/etc/sysfs.conf`` configuration file to do this.
+You will also want to turn off any hyperthreading for cores you use for benchmarking.  You can do
+this in the bios or by something like:
+
+```
+$ echo "0" > /sys/devices/system/cpu/cpu[n]/online
+```
+
+for each hyperthread core you want running - you can look in `/proc/cpuinfo` for pairs of cpus
+that have the same core id turning off one of the pair.  These will need to be done on every boot.
+You can use the sysfsutils package and its `/etc/sysfs.conf` configuration file to do this
+persistently on boot - i.e.:
+
+```
+devices/system/cpu/intel_pstate/no_turbo = 1
+devices/system/cpu/cpu3/online = 0
+devices/system/cpu/cpu5/online = 0
+```
+
+Test Harness Benchmarks
+------------------------
+
+Benchmarks can be run through the test harness (i.e.  using the `run_tests` script) by doing
+e.g. `./run_tests --run speedtests`.  When this is done, the test harness looks for test spec
+files named `speedtests` just like the `tests` files that contain regular moose test details.
+The format for these files is:
+
+```
+[Benchmarks]
+    [benchmark-name]
+        type = SpeedTest
+        input = input-file-name.i
+        cli_args = '--an-arg=1 a/hit/format/cli/arg=foo'
+        # optional:
+        min_runs = 15 # default 40
+        max_runs = 100 # default 400
+        cumulative_dur = 100 # default 60 sec
+    []
+
+    [./benchmark2-name]
+        type = SpeedTest
+        input = another-input-file-name.i
+        cli_args = 'some/cli/arg=bar'
+    []
+
+    # ...
+[]
+```
+
+After being run, benchmark data are stored in a sqlite database (default name
+`speedtests.sqlite`).  When the test harness is run without the `--run speedtests` flag, tests
+described in `speedtests` files are run in *check-only* mode where moose just checks that their
+input files are well-formed and parse correctly without actually running them.
+
 
 Manual/Direct Benchmarks
-++++++++++++++++++++++++++
+--------------------------
 
-This script can also be used to manually list and directly run benchmarks without the
-test harness (for hacking, debugging, etc.).  To do this, the script reads a ``bench.list`` text
+The `[moose-repo]/scripts/benchmark.py` script can be used to manually list and directly run benchmarks without the
```

Comment:
What is `[moose-repo]` for?

Reply:
their moose git repository

Reply:
We usually use ~/projects/moose for this type of thing.